### ISSUE TITLE
[Feature request] Make Ctrl-i of Notepad++ work like Alt-q in Emacs to reformat a paragraph

### ISSUE THREAD

**Issue by fkbreitl** [notepad-plus-plus/notepad-plus-plus#4177]
This is a feature request to make `Ctrl-i` in Notepad++ work like `Alt-q` in Emacs to reformat a paragraph.

Currently `Ctrl-i` already breaks long lines according to the number of columns in the preferences.

However it is missing the functionality to merge shorter lines separated by line breaks. But this is very important for getting edited text back in shape.

For example, remove some words from the text below and format it to a 80 column with:

> Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.

This should be possible by just pressing `Ctrl-i`.

**Comment by alankilborn** [comment]
>it is missing the functionality to merge shorter lines separated by line breaks

_Join Lines_ feature, default keycombo Ctrl+J

Recommend CLOSURE of this issue.

**Comment by fkbreitl** [comment]
Additionally pressing Ctrl-j is not what I requested. It is unnecessarily delaying work.
I requested this functionality to happen by solely pressing Ctrl-i.
Until this is implemented this request should be open.
Closing it was a mistake.
Please reopen!

**Comment by ArkadiuszMichalski** [comment]
For op example text I get this after spiting:
![image](https://user-images.githubusercontent.com/2730894/182848631-0255a419-0a8f-4372-97d5-59662a2ed22e.png)

This 3rd line looks strange, why is so short?

**Comment by fkbreitl** [comment]
Yes, there are problems with Ctrl-i. It should merge it, but doesn't.

**Comment by alankilborn** [comment]
>>This 3rd line looks strange, why is so short?

>Yes, there are problems with Ctrl-i. It should merge it, but doesn't.

Since splitting of lines is a Scintilla function (see [HERE](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/70660eb608eea7a9c85a4bfb08761ffd6bfbc887/PowerEditor/src/NppCommands.cpp#L1760) and [HERE](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/70660eb608eea7a9c85a4bfb08761ffd6bfbc887/PowerEditor/src/NppCommands.cpp#L1773) ), complaints about questionable results should be directed to the Scintilla project.

**Comment by ArkadiuszMichalski** [comment]
> > > This 3rd line looks strange, why is so short?
> 
> > Yes, there are problems with Ctrl-i. It should merge it, but doesn't.
> 
> Since splitting of lines is a Scintilla function (see [HERE](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/70660eb608eea7a9c85a4bfb08761ffd6bfbc887/PowerEditor/src/NppCommands.cpp#L1760) and [HERE](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/70660eb608eea7a9c85a4bfb08761ffd6bfbc887/PowerEditor/src/NppCommands.cpp#L1773) ), complaints about questionable results should be directed to the Scintilla project.

I checked SciTE and Notepad2 and they both don't make such short lines in the example above.

**Comment by alankilborn** [comment]
>I checked SciTE and Notepad2 and they both don't make such short lines in the example above.

I'd guess there is either a difference in Scintilla version then, or a difference in the parameter sent to SCI_SPLITLINES.  Since the parameter seems correct for some of the lines, I'd still blame Scintilla for creating the too-short line.

**Comment by ArkadiuszMichalski** [comment]
Both send 0 (same as NPP), maybe Scintilla version... will see after update.

**Comment by alankilborn** [comment]
>Both send 0 (same as NPP)

I thought from your N++ example that it would be non-zero...because you have an edge set (in your screenshot).

**Comment by ArkadiuszMichalski** [comment]
For easier comparison between programs I delete edge setting (which means the width of the window):
https://www.scintilla.org/ScintillaDoc.html#SCI_LINESSPLIT
Unfortunately Notepad ++ still makes a one line too short.

**Comment by alankilborn** [comment]
>looks like enable the option Show All Characters affects the split.

That seems wrong, but again a Scintilla problem.

**Comment by fkbreitl** [comment]
@alankilborn 
You might be right about the script. But I can't test it. I just used to be an occasional user of Notepad++ and stopped because of some shortcomings like the one mentioned here.
If this script does indent the block apparently correct, then lets assume it does what I want and implement it.
